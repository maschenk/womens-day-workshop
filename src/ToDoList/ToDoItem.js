import React from 'react';

const ToDoItem = ({ item: { itemText, checked }, onCheck, index }) => {
  const handleCheck = () => {
    onCheck(index);
  };

  return (
    <li data-testid="to-do-item">
      <input
        type="checkbox"
        checked={checked}
        onChange={handleCheck}
        data-testid="item-checkbox"
      />{' '}
      {itemText}
    </li>
  );
};

export default ToDoItem;
