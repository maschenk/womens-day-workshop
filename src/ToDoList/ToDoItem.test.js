import { render, fireEvent } from '@testing-library/react';
import ToDoItem from './ToDoItem';

const onCheckMock = jest.fn();
const itemMock = { itemText: 'test item', checked: false };

test('should render', () => {
  const { container } = render(
    <ToDoItem item={itemMock} onCheck={onCheckMock} index={0} />
  );
  expect(container).toMatchSnapshot();
});

test('should be checkable', async () => {
  const component = render(
    <ToDoItem item={itemMock} onCheck={onCheckMock} index={0} />
  );
  const checkBox = component.getByTestId('item-checkbox');

  await fireEvent.click(checkBox);

  expect(onCheckMock).toHaveBeenCalled();
});
