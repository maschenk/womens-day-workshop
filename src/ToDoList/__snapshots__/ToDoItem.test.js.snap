// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <li
    data-testid="to-do-item"
  >
    <input
      data-testid="item-checkbox"
      type="checkbox"
    />
     
    test item
  </li>
</div>
`;
